### ISSUE TITLE
Integrate Android build into celix develop branch

### ISSUE THREAD

**Issue by bpetri** [bpetri/nativeLoad#29]
Right now, the Apache Celix Android features uses ifdefs to define which code parts can be used or not.  This is not the preferred way to to this. 

Refactor the celix code to 
- use a cmake toolchain file for cross compiling
- generate a config.h file to set up dedicated defines for not supported functionality
- check whether the not supported functionality can be extracted into separate code, so we can use the defines in the CMake Configuration but not in the source code itself.

**Comment by marcojansen** [comment]
I've generated a config.h to check for ifaddrs.h ( this is used for retrieving ip and not available for android). I've extracted the method getIpAddress to a seperate file which gets included if ifaddrs.h is available, there are still two #ifdef HAVE_IFADDRS_H in the rsa dfi one  around the check if (ip == null .... ) and one around the include for the h file containing the getIpAddress. 

I've also looked into using a cmake toolchain file. This still requires you to have the correct crosscompiled libraries and  build toolchains. I don't think this is easier to use than the shell scripts I currently have ( extracting correct toolchains from android-ndk, downloading and crosscompiling libraries needed for celix, building celix and building demonstrator.).
